src/efi.rs

```diff
@@ -139,17 +139,28 @@ impl Efi {
             log::debug!("Not booted via EFI, skipping firmware update");
             return Ok(());
         }
-        // Read /etc/os-release
-        let release: OsRelease = OsRelease::new()?;
-        let product_name: &str = &release.name;
+        let product_name = get_product_name()?;
         log::debug!("Get product name: {product_name}");
         assert!(product_name.len() > 0);
         // clear all the boot entries that match the target name
-        clear_efi_target(product_name)?;
-        create_efi_boot_entry(device, espdir, vendordir, product_name)
+        clear_efi_target(&product_name)?;
+        create_efi_boot_entry(device, espdir, vendordir, &product_name)
     }
 }
 
+#[context("Get product name")]
+fn get_product_name() -> Result<String> {
```

Comment:
Should this change to `fn get_product_name(root: &Dir)` to pass root dir as parameter?

Reply:
Yeah, but not a blocker, just a nice to have IMO.

Reply:
Thanks, will update in the later PR.